code/SQLite3Query.php

```diff
@@ -53,18 +53,13 @@ public function seek($row)
         return $result;
     }
 
-    /**
-     * @todo This looks terrible but there is no SQLite3::get_num_rows() implementation
-     */
     public function numRecords()
     {
-        $this->handle->reset();
-        $c=0;
-        while ($this->handle->fetchArray()) {
-            $c++;
+        $columns = $this->handle->numColumns();
```

Comment:
Columns are not rows? And a record is a row... right? unless SQLight has some weird interpretation of data tables...

Reply:
oh indeed, not sure how I missed that... anyway, the current code is throwing some strange errors (currently, i'm trying to run a test suite using sqlite in memory, and it's not working so great do to create table queries that are causing errors) so there is something to be done, but certainly not my proposal. I'll close this and make another pr with a properly tested solution if I find any.